### PR TITLE
support escape character in cell

### PR DESCRIPTION
**Pull request by adereq**

### CODE REVIEW

File: src/parser.rs [hunk 1]
```diff
@@ -393,48 +409,40 @@ impl<'a> Parser<'a> {
...
 
     // returns slice from the next character to `ch`
     // the result is exclusive (does not contain `ch`), but the functions consumes `ch`
+    // when `ch` is preceded by `\` character, it ignores this occurence of `ch`
     // None is returned when the input is empty or when `ch` is the next character
     // Examples:
     // Parser::new("foObar").slice_to_exc('b') == Some("foO"), self.cur.next() == (4, 'a')
     // Parser::new("foObar").slice_to_exc('f') == None,        self.cur.next() == (1, 'o')
+    // Parser::new("fo\ObaOr").slice_to_exc('O') == Some("fo\Oba"), self.cur.next() == (7, 'r')
     fn slice_to_exc(&mut self, ch: char) -> Option<&str> {
```
Comment (Patryk27) [inline src/parser.rs:431]: considering that this function is also called for reading strings, we could relatively easily add support for parsing escapes there as well 🙂 
```
foo = "hello \" world!"
```
```patch
diff --git a/src/ion/display.rs b/src/ion/display.rs
index e2ebf82..c20cb9c 100644
--- a/src/ion/display.rs
+++ b/src/ion/display.rs
@@ -36,7 +36,18 @@ impl fmt::Display for Section {
 impl fmt::Display for Value {
     fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
         match *self {
-            Value::String(ref v) => v.fmt(f),
+            Value::String(ref v) => {
+                for c in v.chars() {
+                    match c {
+                        '\\' => write!(f, "\\\\")?,
+                        '\n' => write!(f, "\\n")?,
+                        '\"' => write!(f, "\\\"")?,
+                        c => write!(f, "{c}")?,
+                    }
+                }
+
+                Ok(())
+            }
             Value::Integer(ref v) => v.fmt(f),
             Value::Float(ref v) => v.fmt(f),
             Value::Boolean(ref v) => v.fmt(f),
diff --git a/src/parser.rs b/src/parser.rs
index aa9de2c..c866b5c 100644
--- a/src/parser.rs
+++ b/src/parser.rs
@@ -303,7 +303,14 @@ impl<'a> Parser<'a> {
 
     fn finish_string(&mut self) -> Option<Value> {
         self.cur.next();
-        self.slice_to_exc('"').map(|s| Value::String(s.to_string()))
+
+        self.slice_to_exc('"')
+            .map(|s| {
+                s.replace("\\\\", "\\")
+                    .replace("\\n", "\n")
+                    .replace("\\\"", "\"")
+            })
+            .map(Value::String)
     }
 
     fn keyval_sep(&mut self) -> bool {
diff --git a/tests/data/test.ion b/tests/data/test.ion
index 74ccc0c..4b75801 100644
--- a/tests/data/test.ion
+++ b/tests/data/test.ion
@@ -1,6 +1,6 @@
 [dict]
 ary = [ "a", "b", 3 ]
-first = "first"
+first = "hello \\ hello \n hello \""
 # comment
 second ="another"
 bool = true
diff --git a/tests/expected/test.ion b/tests/expected/test.ion
index 41c2557..889a5bd 100644
--- a/tests/expected/test.ion
+++ b/tests/expected/test.ion
@@ -1,7 +1,7 @@
 [dict]
 ary = [ "a", "b", 3 ]
 bool = true
-first = "first"
+first = "hello \\ hello \n hello \""
 second = "another"
 
 [mixed]
```

File: src/ion/section.rs [hunk 2]
```diff
@@ -137,7 +131,7 @@ mod tests {
 
             #[test]
             fn it_works_on_ref_section() {
-                let ion = ion!(
+                let ion = crate::ion!(
```
Comment (Patryk27) [inline src/ion/section.rs:134]: nit: can't this be just imported at the top here? (i.e. near `mod tests {`) :eyes:

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,8 @@
 # Changelog
 
+## 0.8.9
+- Support escape `\` character when creates cell
```
Comment (Patryk27) [inline CHANGELOG.md:4]: nit: `when reading cells and strings`

File: src/parser.rs [hunk 4]
```diff
@@ -444,43 +452,40 @@ impl<'a> Parser<'a> {
...
-            desc: message.to_owned()
+        self.errors.push(ParserError {
+            lo,
+            hi,
+            desc: message.to_owned(),
         });
     }
 }
 
 fn is_digit(c: char) -> bool {
```
[the remark this answers is not in the document]
Comment (adereq) [reply src/parser.rs]: fixed

File: src/parser.rs [hunk 5]
```diff
@@ -331,15 +342,18 @@ impl<'a> Parser<'a> {
     }
 
     fn cell(&mut self) -> String {
-        self.ws();
-        self.slice_to_exc('|').map(str::trim_end).unwrap_or("").to_owned()
+        self.whitespace();
+        self.slice_to_exc('|')
+            .map(str::trim_end)
+            .unwrap_or("")
+            .replace('\\', "")
```
[the remark this answers is not in the document]
Comment (adereq) [reply src/parser.rs]: fixed

File: src/parser.rs (hunk 1 above)
Comment (adereq) [reply src/parser.rs:439]: fixed

File: src/ion/section.rs (hunk 2 above)
Comment (adereq) [reply src/ion/section.rs]: fixed

File: CHANGELOG.md (hunk 3 above)
Comment (adereq) [reply CHANGELOG.md]: fixed

**Pull request merged**